### PR TITLE
added back button to the visualization.

### PR DESCRIPTION
**Pull request by sebastian-nehrdich**
This is the back-button feature. @ayya-vimala I didn't add Hubert to the reviewers but if you think it's better to have him have a look at the PR before merge then I will ask him!

### CODE REVIEW

File: src/views/visual-view.js [hunk 1]
```diff
@@ -21,7 +18,8 @@ class VisualView extends BaseView {
       labels: { type: Object },
       colors: { type: Array },
       options: { type: Object },
-      oldUrl: { type: String },
+	oldUrl: { type: String },
```
Comment (ayya-vimala) [inline src/views/visual-view.js]: Added tab in front of oldUrl does not seem to be needed.

File: src/views/visual-view.js [hunk 2]
```diff
@@ -32,7 +30,8 @@ class VisualView extends BaseView {
     this.leftSelected = '';
     this.rightSelected = '';
     this.viewData = '';
-    this.oldUrl = '';
+      this.oldUrl = '';
+      this.backUrls = [];
```
Comment (ayya-vimala) [inline src/views/visual-view.js]: Again, somewhere it adds tabs

File: src/views/visual-view.js [hunk 3]
```diff
@@ -152,6 +152,13 @@ class VisualView extends BaseView {
           </vaadin-list-box>
         </template>
       </vaadin-select>
+          <vaadin-button
+            theme="contrast primary small"
+            class="back-button element-hidden"
+            style="display:none"
```
Comment (ayya-vimala) [inline src/views/visual-view.js]: It seems to me that you do the same thing twice here: add a class 'element-hidden' and put a 'display-none' in the elements style. I suggest to just one of the two. Either add or remove the style or add or remove the class.

File: src/views/visual-view.js [hunk 4]
```diff
@@ -198,11 +215,21 @@ class VisualView extends BaseView {
...
+	// we do not want to show the backButton when the visualization is on top-level - this works partly now, but it could be better.
+	if(this.backUrls.length > 0){
+	    document.getElementsByClassName('back-button')[0].style.display = '';
+	}
+	else{
+	    document.getElementsByClassName('back-button')[0].style.display = 'none';
+	}
+	console.log("URL",url);
+	console.log("OLDURL",this.oldUrl);
+	console.log("BACKURLS",this.backUrls);
```
Comment (ayya-vimala) [inline src/views/visual-view.js]: Maybe remove the console messages here or just leave the current one with a more descriptive message to say what it is doing.

File: src/views/visual-view.js [hunk 5]
```diff
@@ -212,6 +239,7 @@ class VisualView extends BaseView {
       win.focus();
       return;
     }
+	this.backUrls.push(url);
```
Comment (ayya-vimala) [inline src/views/visual-view.js]: too many tabs again

File: src/views/visual-view.js (hunk 1 above)
Comment (ayya-vimala) [reply src/views/visual-view.js]: This is a bit strange because in the review it shows with an extra tab but here it doesn't ... :-(

Comment (sebastian-nehrdich) [reply src/views/visual-view.js]: I have no idea about the tabs, probably it's just a different preference of my editor. There is no special intention behind this on my side. I will try to change the settings so we don't get this so much in the fturue.

File: src/views/visual-view.js (hunk 3 above)
Comment (sebastian-nehrdich) [reply src/views/visual-view.js]: OK! Thanks for pointing this out, style="display:none"   should be enough.

File: src/views/visual-view.js (hunk 1 above)
Comment (ayya-vimala) [reply src/views/visual-view.js]: Yes, just leave those. It is not important.

File: src/views/visual-view.js (hunk 4 above)
Comment (sebastian-nehrdich) [reply src/views/visual-view.js]: I will remove them.

**Pull request merged**